src/shogun/lib/sg_type_traits.h

```diff
@@ -0,0 +1,71 @@
+/*
```

Comment:
Nitpicking a bit. Maybe we can put the concepts in a separate `sg_concepts.hpp` file or rename to something else that includes traits and concepts.

Reply:
Completely agreed - Will do so.